### ISSUE TITLE
Is ctdTrim(method="sbe") working as expected?

### ISSUE THREAD

**Issue by j-harbin** [dankelley/oce#2012]
Hi All,

I'm new to processing CTD data, so this could very well be me misunderstanding the `ctdTrim()` documentation. I have the following code 

```
library(oce)
f <- read.oce("SBE19plus_01906009_2019_04_11_0002.cnv")
par(mfrow=c(1,2))
plotScan(f)
s <- ctdTrim(f, method='sbe', parameters=list(minSoak=1, maxSoak=18))
plotScan(s)
```
which produces the following:

![Screen Shot 2022-11-03 at 10 44 05 AM](https://user-images.githubusercontent.com/58750538/199736971-d744a7dc-b7f8-489b-a2f0-675e8fd68b77.png)

Based on the documentation for `method="sbe"`, indicating " a method similar to that described in the SBE Data Processing manual is used to remove the "soak" period at the beginning of a cast", I would have expected it to only remove the data until approx 1000 m (ie. not the upward cast as well). Any input on this would be great :)

**Comment by j-harbin** [comment]
By 1000 m, I mean 1000 scans .. :)

This makes sense. I think based on the definition, I thought it was only going to remove the first 800 or so scans due to the soak, and then I would need to do:

`ctdTrim(f, method="downcast")` to remove the upcast.

Thanks for the clarification, Clark. I'll close this issue.

**Comment by richardsc** [comment]
You'll find as you go through the docs on `ctdTrim()` that there are *a lot* of options. It's worth playing around with them to see how they affect a particular profile.

But it's also worth considering that sometimes manually determining and specifying the exact scan range to trim can save you a lot of hassle of trying to get one method to work for a whole range of profiles.

**Comment by dankelley** [comment]
Re the manual method ... I always like to do a cost comparison on this.  An oceanographic vessel costs something like $30K per day.  A section I was looking recently involved 68 profiles over 20 days of sampling. That works out to nearly $9000 per profile.  Think about that in terms of the time it would take a technician to plot the data and decide where to cut it for a downcast.  Maybe that time would be 0.5 hours, for a cost of let's say $20 per profile.

So the ratio of analysis cost to acquisition cost is about 1 part in 450 parts.  The analyst's time is essentially zero compared to the overall cost of the data.  Therefore an analyst should not feel bad about taking 0.5 hours to look at a profile.  The same goes for QC.

My 2 cents worth.

**Comment by j-harbin** [comment]
I agree with manually scanning and trimming. I am going to reopen this issue because I do think there is further discussion to be had regarding `ctdTrim(method="sbe")`. See the following code for a separate file (note: majority of the files have a similar outcome)

```
library(oce)
f <- read.oce("SBE19plus_01906009_2019_04_11.cnv")
par(mfrow=c(1,2))
plotScan(f)
mtext("Raw File", line=0, cex=1)
s <- ctdTrim(f, method='sbe', parameters=list(minSoak=1, maxSoak=18))
plotScan(s)
mtext("Soaked Removed", line=0, cex=1)
```


![Screen Shot 2022-11-03 at 1 05 30 PM](https://user-images.githubusercontent.com/58750538/199772713-f8c14e2e-2e2a-400b-9baa-a100ddd4e308.png)

Where in this case I would have expected everything until about 3500 scans to be removed.

**Comment by dankelley** [comment]
@richardsc will be more useful on the policy decision that I could be.  That is an interesting looking set of data.  The descent to 10m (or whatever that is) seems to be deliberate.  And kind of odd. That seems pretty deep, but I don't know what sort of policy is being used by the technicians.

@j-harbin can you email me the datafile? I'd like to take a brief look (between grading this and that and the other..)

**Comment by dankelley** [comment]
Thanks for the file.  Well, it's an odd file.  They kept the ctd very near 10m for a long time -- 10 minutes I think -- as shown below.   But the data show very fresh water, well below a freezing point.  Maybe @richardsc can explain what's going on, and make suggestions for how to proceed, and whether oce needs to be modified.
```R
library(oce)
png("jh_%d.png")
f <- "SBE19plus_01906009_2019_04_11.cnv"
d <- read.oce(f)
plot(d, eos="unesco")
plotScan(d, type="p", cex=0.05, xtype="time")
p <- d[["pressure"]]
plateau <- 10 < p & p < 12
ds <- subset(d, plateau)
plotScan(ds, xtype="time", type="p", cex=0.5)
```

**jh_1**

![jh_1](https://user-images.githubusercontent.com/99469/199800433-c06991cb-7e55-4b56-8f31-66920e2ae929.png)

**jh_2**

![jh_2](https://user-images.githubusercontent.com/99469/199800464-4ed51be0-1024-40cb-9b12-eeaf9735b2cb.png)

**jh_3**

![jh_3](https://user-images.githubusercontent.com/99469/199800492-335d924c-8dc0-4ef8-b770-ac1d44d6a0f7.png)

**Comment by clayton33** [comment]
@dankelley a soak at 10m is what we routinely do. I'm not sure if it's a standard that's followed everywhere, but it seems to be a good depth to send the rosette down to in order for sensors to equalize. What does the profile look like after you trim it ? They may have left it there for a period of time because the water in the tubing froze (which I would be terrified if that's the case).

**Comment by dankelley** [comment]
@clayton33 thanks, that's good to know.  The plot (a couple of comments back) suggests very fresh water at sub-freezing temperature. The actual profile data seem ok.  I'm not really following this discussion; it's J and C mainly. We had talked about maybe a 3-way zoom.  Maybe 4-way.

**Comment by richardsc** [comment]
I know we're going to meet to chat through some of this (which is still valuable for @j-harbin), but just so you can see what kind of difference noodling with the parameters can make, I got the trim to work on the first try 😄 . The only real difference is in the `minSoak` parameter, which I choose to be larger than 1 dbar to make sure no weirdness at the initial "dip" would throw off the algorithm.

```r
library(oce)
d <- read.oce('SBE19plus_01906009_2019_04_11.cnv')
dt <- ctdTrim(d, method='sbe',
              parameters = list(minSoak=5, maxSoak=15))
plotScan(dt)
plotTS(dt)
```
<img width="675" alt="image" src="https://user-images.githubusercontent.com/233584/199863291-f6d5abf6-f23d-4034-a3fd-40398596aaa6.png">
<img width="683" alt="image" src="https://user-images.githubusercontent.com/233584/199863323-9100a887-dd7e-4bba-922c-f39db0bfc8c5.png">

**Comment by dankelley** [comment]
I knew @richardsc would figure that out.  I think we still ought to meet on Monday, partly to discuss whether the default `minSoak` and `maxSoak` ought to be altered, and whether better efforts should be made to find the soak (perhaps based on number of samples at a fixed pressure ... if soaking for a really long time is common, that could work).

Also of interest (to me anyway) is the variation of S and T during the soak period. I plotted those things up, and you can see them below.  I find the S signal to be quite pretty interesting.  I should plot the pre-soak also, I think, before the meeting.

![jh_d4_1](https://user-images.githubusercontent.com/99469/199948087-fb5c869b-2659-4646-8716-cc4ee72aa553.png)

![jh_d4_2](https://user-images.githubusercontent.com/99469/199948218-b37f884e-24e5-469a-944e-8558c731cc7c.png)

![jh_d4_3](https://user-images.githubusercontent.com/99469/199948248-8a8b8a69-56f5-4d38-8d1c-1f8eb0976f09.png)

**Comment by richardsc** [comment]
Nice plots Dan -- that's a good demonstration of *why* we do the soak! To make sure that the instruments have equilibrated to a temperature close to ambient to minimize any thermal inertia issues once the profile starts. You can see that 10 minutes looks like it's only barely enough time!

**Comment by dankelley** [comment]
@richardsc yup, this is why I love to see the raw data.  Quarter-second: pretty sweet.  (PS. the Julian-Day column just barely has enough digits to make succesive samples distinct!)

**Comment by dankelley** [comment]
I'm not even sure that the soak period has been barely enough...maybe the unit is just sampling natural variability in a mixing fluid...

**Comment by j-harbin** [comment]
I'm going to close this issue as myself, @dankelley, and @annieHow have had in depth conversations about this. We are in the process of testing new methods to maybe eventually improve this method.
